hedera-node/src/main/java/com/hedera/services/sigs/sourcing/SigMapPubKeyToSigBytes.java

```diff
@@ -53,15 +50,18 @@
 
 	@Override
 	public byte[] sigBytesFor(byte[] pubKey) throws KeyPrefixMismatchException {
-		List<byte[]> matchingSigs = sigMap.getSigPairList()
-				.stream()
-				.filter(sp -> beginsWith(pubKey, sp.getPubKeyPrefix().toByteArray()))
-				.map(this::sigBytesFor)
-				.collect(toList());
-		if (matchingSigs.size() > 1) {
-			throw new KeyPrefixMismatchException("Source signature map is ambiguous for given public key!");
+		byte[] sigBytes = EMPTY_SIG;
+		for (SignaturePair sigPair : sigMap.getSigPairList()) {
+			final byte[] pubKeyPrefix = sigPair.getPubKeyPrefix().toByteArray();
+			final int length = pubKeyPrefix.length;
```

Comment:
For each iteration, these two values will be re-assigned. Is it ok to make them final?